### PR TITLE
Add redo and undo feature

### PR DESCRIPTION
**Pull request by adam07018**
v1.0 for checkstyle

### CODE REVIEW

**Comment by adam07018** [comment]
### Commit Feature

1. A change is going to be made upon current AddressBook, eg. setAddressBook. deletePerson, addPerson. Call it function x
2. ModelManager#function x is invoked. After change was made, VersionedAddressBook#commit will be called.
3. Record down current version as the latest AddressBook in history.

**Comment by adam07018** [comment]
### Undo Feature

1. User type in "undo"
2. Calls **LogicManager#execute** 
3. Pass to **addressBookParser#parseCommand**, return a undoCommand back to **LogicManager#execute** 
4. Calls **UndoCommand#execute**. Check if we can undo by calling **model#canUndo**. If can, call **model#undo**
5. Calls **versionedAddressBook#undo** which returns reference to undo version of Addressbook.
6. Calls **Model#setAddressBook** (Overload function). Reset current AddressBook to clone of undo version of Addressbook.

**Comment by adam07018** [comment]
### Redo Feature
Similar to Undo

**Comment by adam07018** [comment]
@pangrwa @tituswe Friends, can help me assign the milestone?

**Review by tituswe — COMMENTED** [review]
LGTM. But wait on refactoring PR, will have to deconflict.

File: src/test/java/seedu/address/model/VersionedAddressBookTest.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    void redo() {
+    }
+
+    @Test
+    void canUndo() {
+    }
+
+    @Test
+    void canRedo() {
+    }
```
Comment (tituswe) [inline src/test/java/seedu/address/model/VersionedAddressBookTest.java:28]: Likewise for here, we should try to complete this by v1.2b

**Review by tituswe — APPROVED** [review]
Let's finish the test cases by v1.2b. Otherwise, looks great!

**Pull request merged**